crates/exex/exex/src/manager.rs

```diff
@@ -241,31 +250,176 @@ impl<Node: Unpin> Stream for ExExNotifications<Node> {
 /// A stream of [`ExExNotification`]s. The stream will only emit notifications for blocks that are
 /// committed or reverted after the given head.
 #[derive(Debug)]
-pub struct ExExNotificationsWithHead<Node> {
+pub struct ExExNotificationsWithHead<Node: FullNodeComponents> {
     #[allow(dead_code)]
     components: Node,
     notifications: Receiver<ExExNotification>,
     head: ExExHead,
+    /// The backfill job to run before consuming any notifications.
+    backfill_job: Option<StreamBackfillJob<Node::Executor, Node::Provider, Chain>>,
+    /// Whether to wait for the node head to be at the same height as the ExEx head, and then
+    /// call the [`Self::heal`].
+    wait_for_head: bool,
```

Comment:
renamed the field

> I assume we can derive the same functionality by using `Option<Notifaction>` which is the most recent notification we received from the stream and compare the exexhead against this, or maybe as an additional field

we can do that but unsure what's the benefit because we will set this field to `Some(notification)` inside the `Stream::poll_next` and immediately use it to start healing process, if needed